Add Kind cluster setup (#1538)
kyma-project/test-infra: kind/README.md

> # kind Environment

Reviewer: Maybe add \` \` around "kind", to make it look like a project name

## Overview

> kind is a tool for running local Kubernetes cluster with nodes deployed as Docker containers. It can be deployed as a multi-node container what gives as ability to test in a environment close to production cluster.

Reviewer: same here

## Directory structure

The `kind` folder contains a set of configuration files for the kind cluster.

Its structure looks as follows:

```
  ├── cluster                  # kind cluster configuration
        └── resources          # Default resources applied during cluster setup
  ├── config                   # Kyma overrides for kind
  ├── scripts                  # Scripts used for kind environment
  └── KIND_KUBERNETES_VERSION  # File with Kubernetes version used for cluster
```

## Deploy Kyma cluster on kind

To deploy Kyma on kind, execute:

```bash
./scripts/install-kyma.sh
```

To list available options, execute:

```bash
./scripts/install-kyma.sh --help
```

> It may require more resources for Docker, adjust it in Docker preferences (cpu and memory)

Reviewer 2: I suggest to be more precise here, what values are recommended to set in docker preferences

Reviewer 3: ```

> **TIP:** You may have to adjust CPU and memory settings for Docker. Use the recommended values of 6 CPU and 12 GB. Adjust these values in Docker preferences. 

```

## Build Kubernetes image

> To build a Kubernetes image that is used in Kind, script `scripts/build-kubernetes-image.sh` is used. That script is designed to be used on Prow pipeline.

Reviewer: it's Kind or kind? Please be consistent

> To build a Kubernetes image that is used in Kind, script `scripts/build-kubernetes-image.sh` is used. That script is designed to be used on Prow pipeline.

Reviewer 3: ```suggestion
Run the `scripts/build-kubernetes-image.sh` script to build a Kubernetes image used in `kind`. That script is specifically designed for the Prow pipeline.
```

> For creating your own Kubernetes image, follow https://kind.sigs.k8s.io/docs/user/quick-start/#building-images.

Reviewer 3: ```suggestion
To create your own Kubernetes image, follow [this](https://kind.sigs.k8s.io/docs/user/quick-start/#building-images) instruction.
```
